### ISSUE TITLE
etcd member list / member remove fails on healthy cluster

### ISSUE THREAD

**Issue by mariusgrigaitis** [coreos/etcd#2834]
etcd 2.0.10

1) Created a cluster of 4 machines, all were healthy
2) Member list
```
4841c25d090e96fe: name=ip-10-128-1-31 peerURLs=http://10.128.1.31:2380 clientURLs=http://10.128.1.31:2379,http://10.128.1.31:4001
4d69e7a82152c25d: name=ip-10-128-1-209 peerURLs=http://10.128.1.209:2380 clientURLs=http://10.128.1.209:2379,http://10.128.1.209:4001
5e2dc7d2a0342ecb: name=ip-10-128-1-249 peerURLs=http://10.128.1.249:2380 clientURLs=http://10.128.1.249:2379,http://10.128.1.249:4001
f4f9314423ec9cf4: name=soad-cluster-node1 peerURLs=http://172.21.3.1:2380 clientURLs=http://172.21.3.1:2379,http://172.21.3.1:4001
```
3) Then killed the first member (turned off the machine)
4) 
```
$ etcdctl cluster-health
cluster is healthy
member 4841c25d090e96fe is unhealthy
member 4d69e7a82152c25d is healthy
member 5e2dc7d2a0342ecb is healthy
member f4f9314423ec9cf4 is healthy
```
5)
```
admin@ip-10-128-1-249:~$ etcdctl member list
context deadline exceeded
admin@ip-10-128-1-249:~$ etcdctl member remove 4841c25d090e96fe
Recieved an error trying to remove member 4841c25d090e96fe: context deadline exceeded
```

**Comment by mariusgrigaitis** [comment]
Sometimes error when removing member is: ```Error while verifying ID against known members: context deadline exceeded```

And sometimes member list succeeds:

```
admin@ip-10-128-1-249:~$ etcdctl member list
context deadline exceeded
admin@ip-10-128-1-249:~$ etcdctl member list
4841c25d090e96fe: name=ip-10-128-1-31 peerURLs=http://10.128.1.31:2380 clientURLs=http://10.128.1.31:2379,http://10.128.1.31:4001
4d69e7a82152c25d: name=ip-10-128-1-209 peerURLs=http://10.128.1.209:2380 clientURLs=http://10.128.1.209:2379,http://10.128.1.209:4001
5e2dc7d2a0342ecb: name=ip-10-128-1-249 peerURLs=http://10.128.1.249:2380 clientURLs=http://10.128.1.249:2379,http://10.128.1.249:4001
f4f9314423ec9cf4: name=soad-cluster-node1 peerURLs=http://172.21.3.1:2380 clientURLs=http://172.21.3.1:2379,http://172.21.3.1:4001
```

**Comment by mariusgrigaitis** [comment]
It's possible to write / read values to such cluster.

**Comment by mariusgrigaitis** [comment]
Tested with 2.1.0-alpha.0. Problem persists.

**Comment by yichengq** [comment]
@mariusgrigaitis try to set `--peers` for etcdctl? Moreover, could you open `--debug` when use etcdctl?
Do you have the log of etcd processes?

**Comment by xiang90** [comment]
related to https://github.com/coreos/etcd/pull/2906

**Comment by xiang90** [comment]
@mariusgrigaitis We have fixed this on current master branch. Please give it a try if you want. We will release a new alpha with the fix this week. Thanks for reporting!

**Comment by bmccandless** [comment]
Hi Xiang,
 When do you think the next new alpha will be release?   I'd like to start
using this fix, and would rather have it in an official build rather than
the latest master.
   Thanks you,
      Brian

**Comment by xuwang** [comment]
I can reproduce the  problem on beta 695.0.0, with 3 nodes cluster: removing a dead node rendered the cluster un-healthy. 

The workaround to recover the cluster is to restart etcd2 on one of the remaining two nodes. Based on status log, the restarted node elected the other node as leader, become a follower, and removed the dead node successfully. 
```
ip-10-0-2-101 core # systemctl stop etcd2
Job for etcd2.service canceled.
ip-10-0-2-101 core # systemctl start  etcd2
ip-10-0-2-101 core # systemctl status  etcd2
 etcd2.service - etcd2
   Loaded: loaded (/usr/lib64/systemd/system/etcd2.service; static; vendor preset: disabled)
  Drop-In: /run/systemd/system/etcd2.service.d
           └─10-oem.conf, 20-cloudinit.conf
   Active: active (running) since Sat 2015-06-13 00:42:34 UTC; 24s ago
 Main PID: 30388 (etcd2)
   Memory: 5.8M
   CGroup: /system.slice/etcd2.service
           └─30388 /usr/bin/etcd2

Jun 13 00:42:34 ip-10-0-2-101.us-west-2.compute.internal etcd2[30388]: 2015/06/13 00:42:34 etcdserver: snapshot count = 10000
Jun 13 00:42:34 ip-10-0-2-101.us-west-2.compute.internal etcd2[30388]: 2015/06/13 00:42:34 etcdserver: advertise client URLs = http://10.0.2.101:2379
Jun 13 00:42:34 ip-10-0-2-101.us-west-2.compute.internal etcd2[30388]: 2015/06/13 00:42:34 etcdserver: loaded cluster information from store: etcd_a_01=http://10.0.2.51:2380,etcd_b_01=http://10.0.2.101:2380,etcd_c_01=http://10.0.2.151:2380
Jun 13 00:42:34 ip-10-0-2-101.us-west-2.compute.internal etcd2[30388]: 2015/06/13 00:42:34 etcdserver: restart member b8fc54484fbc9fd in cluster c34c9c8ca3473db8 at commit index 72320
Jun 13 00:42:34 ip-10-0-2-101.us-west-2.compute.internal etcd2[30388]: 2015/06/13 00:42:34 raft: b8fc54484fbc9fd became follower at term 311
Jun 13 00:42:34 ip-10-0-2-101.us-west-2.compute.internal etcd2[30388]: 2015/06/13 00:42:34 raft: newRaft b8fc54484fbc9fd [peers: [b8fc54484fbc9fd,3613ae448e357208,8c60be5a0981a5a7], term: 311, commit: 72320, applied: 70007, lastindex: 72324, lastterm: 311]
Jun 13 00:42:34 ip-10-0-2-101.us-west-2.compute.internal etcd2[30388]: 2015/06/13 00:42:34 raft.node: b8fc54484fbc9fd elected leader 8c60be5a0981a5a7 at term 311
Jun 13 00:42:34 ip-10-0-2-101.us-west-2.compute.internal etcd2[30388]: 2015/06/13 00:42:34 etcdserver: removed member 3613ae448e357208 from cluster c34c9c8ca3473db8
Jun 13 00:42:34 ip-10-0-2-101.us-west-2.compute.internal etcd2[30388]: 2015/06/13 00:42:34 rafthttp: starting client stream to 8c60be5a0981a5a7 at term 311
Jun 13 00:42:35 ip-10-0-2-101.us-west-2.compute.internal etcd2[30388]: 2015/06/13 00:42:35 etcdserver: published {Name:etcd_b_01 ClientURLs:[http://10.0.2.101:2379]} to cluster c34c9c8ca3473db8
```
To add a new node, build the node with the same node name (etcd_a_01) and IP (10.0.2.51), run ` etcdctl member add etcd_a_01  http://10.0.2.51:2380`, and restart the etcd2 on the new node. There seems no data loss.

**Comment by yichengq** [comment]
@bmccandless We just cut 2.1.0-alpha.1 and you can have a try now.
